src/main/java/com/xjeffrose/xio/http/Client.java

```diff
@@ -0,0 +1,104 @@
+package com.xjeffrose.xio.http;
+
+import com.xjeffrose.xio.SSL.SslContextFactory;
+import com.xjeffrose.xio.client.ChannelConfiguration;
+import com.xjeffrose.xio.client.ClientConfig;
+import com.xjeffrose.xio.http.Http1ClientCodec;
+import com.xjeffrose.xio.http.PipelineRequestHandler;
+import com.xjeffrose.xio.http.RawBackendHandler;
+import com.xjeffrose.xio.http.Request;
+import com.xjeffrose.xio.http.Route;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInitializer;
+import io.netty.handler.codec.http.HttpClientCodec;
+import io.netty.handler.ssl.SslContext;
+import java.net.InetSocketAddress;
+import java.util.ArrayDeque;
+import java.util.Queue;
+import io.netty.channel.ChannelHandler;
+import com.xjeffrose.xio.core.XioMessageLogger;
+import com.xjeffrose.xio.core.XioIdleDisconnectHandler;
+import java.util.function.Supplier;
+import java.util.concurrent.Future;
+import io.netty.channel.ChannelFutureListener;
+
+public class Client {
+
+  private final ClientState state;
+  private final Supplier<ChannelHandler> appHandler;
+  Channel channel;
```

Comment:
1) This should be private.

2) I would prefer for this to be final as well but the netty idiom with the `Bootstrap` is to both create the `Channel` and start it doing something. So the two options that I see are:
 a) Have the constructor do more than construct so we can set `final Channel channel`
 b) Defer the construction of `Channel channel` knowing that we will probably never update it again after setting it once.

I'm going with option b) for the time being because it should make testing easier. It may be possible to create a compromise between a) and b) but that would require digging into the Bootstrap class and moving some of it's innards into the `Client` class.